main.go

```diff
@@ -1,11 +1,26 @@
 package main
 
 import (
+	"log"
+	"os"
+
 	"github.com/ivanterekh/go-skeleton/server"
 	"github.com/xlab/closer"
 )
 
 func main() {
-	server.Start(":8080")
+	env := os.Getenv("ENV")
+	if env == "" {
+		env = "unknown"
```

Comment:
1. We must have dev as a deafult value
2. How I should use that in my code? What I asked is to do something like that https://play.golang.com/p/UChkE6_3sRU just organize that code into a package

Reply:
1. dev is set as the default in the Makefile. But in case default dev environment is not provided for some reason this fact should be easy to reveal.
2. I don't realy get how a package which is used like `env.IsProd()` can be better than checking env variable `env=="prod"`, passed to nessasary modules following the DI principles.

Reply:
1. Env package is a kind of pattern and here is a rule "Don't make people think" - here the answer why https://www.youtube.com/watch?time_continue=1&v=70hcbjyJZ-Q
And you must use a default value because your system should be as easier as it can possibly be. For example, our system has tens of parameters but I can run it just with two of them.
Also, imagine that in one day you will mention starting use `"production"` instead of `"prod"` and your changes will take a lot of lines instead of one in the env package.

2. Makefile is not a  part of your application.
How should I build my prod version?
How should I build my staging version?
Can I reuse the code without any changes?
How are you going to run debugger?
How to run tests?
and etc...

3. The idea is that you must create an independent package which encapsulates implementation and provides public functions or constants. And your approach is called spaghetti code.

Reply:
You spend too much time on an easy job. If you don't know how to make something better just do what other people do https://github.com/search?l=Go&q=IsStaging&type=Code
LGTM https://github.com/FeiniuBus/ecosystem/blob/5b068baa6e8818c7acb22d848df117a203a3d85b/hosting/environment.go